enterprise/frontend/src/metabase-enterprise/whitelabel/components/HelpLinkSettings/HelpLinkRadio.tsx

```diff
@@ -0,0 +1,59 @@
+import { t } from "ttag";
+import { useCallback } from "react";
+import { Radio, Stack } from "metabase/ui";
+
+interface Props {
+  setting: { value: string };
+  onChange: (value: string) => void;
+}
+
+export const HelpLinkRadio = ({ setting, onChange }: Props) => {
+  const focusUrlInput = useCallback(async () => {
+    // The url input is rendered only after the setting is saved and returned from the BE
```

Comment:
In theory, we could store some state very up in the component tree or in redux, but it would pollute other components and in this case I think 2 lines of procedural code are cleaner than having a one-off state shared across components just to focus something.

We could also make that getHidden actually hides via css instead of not rendering the settings, but in that case the item would get focused while hidden during the network request.
https://github.com/metabase/metabase/blob/b38ad76666a91915a2f050969ac796f70e63625a/frontend/src/metabase/admin/settings/app/components/SettingsEditor/SettingsSection.tsx#L25-L28

Overall, I'm okay this solution with waitFor but I understand some people may not like it

Reply:
Please remove all of this autofocus implementation and put `autoFocus: true` in the settings instead, since we already have that supported in the settings e.g.
https://github.com/metabase/metabase/blob/08480aac9416ea03ba5ae525f67b27da2ea54112/enterprise/frontend/src/metabase-enterprise/auth/index.js#L88-L95

However I found out that it doesn't work at the moment. I can patch that at the moment [here](https://github.com/metabase/metabase/blob/08480aac9416ea03ba5ae525f67b27da2ea54112/frontend/src/metabase/admin/settings/components/SettingsSetting.jsx#L58C1-L62)

```ts
    const widgetProps = {
      ...setting.getProps?.(setting, settingValues),
      ...setting.props,
      ...this.props,
      ...(setting.autoFocus ? { autoFocus: true } : {}),
    };
```

but it'd be great to ask other engineers about this since I'm not sure since when the `autoFocus` in the settings has stop working.

@ranquild Do you know how we should fix `autoFocus` not working for our settings?

Reply:
@WiNloSt doesn't that also autofocus it also on page load? I only need to autofocus it when "custom" gets selected in the previous setting.
If I open the SAML setup page, the url input gets autofocused and the page scrolls down to it


https://github.com/metabase/metabase/assets/1914270/f71af340-f80b-4f74-8819-af3e64f3786a

And I can't focus it without waitFor because it's not rendered yet, and even if we switched from not rendering it, to hiding it via css (which would require some refactor) it would focus a hidden element until the settings are refreshed, which i don't think it's a good idea

Reply:
Yeah, that's the option for all settings widget whether we want it to be autofocus or not. I feel like conditional autoFocus would be too complex. However, I can see how it could be jarring for the settings page to have the cursor jump to the very bottom input. 

In the case of your video, we should maybe probably remove the autoFocus from the settings itself. But for this Epic. I'd rather not autofocus it or autofocus it all the time if conditional autofocus looks too hacky.

Reply:
Tomorrow I'll draft another alternative, but I'm afraid it won't be much cleaner (if at all)

Reply:
Also, it might be worth to confirm with the product if always autofocusing is viable. Otherwise, I think dropping the autofocus behavior in favor of a simpler code would be ideal.